Share product data in SCC path

rjschwei opened the pull request:
@mcalmer initial testing looks good. Wanted to let you take a look at this as early as possible.

rjschwei committed:
- Fix typo in attribute name for XML representation during data sharing

rjschwei committed:
- Share registered product information with the sibling server
  + At present in the SCC path product registration information is not shared
    with the sibling server. Thus a guest being redirected to the sibling
    will remove it's repositories.

mcalmer commented on www/perl-lib/SMT/RegistrationSharing.pm:
```diff
@@ -243,42 +218,41 @@ sub shareRegistration
+    # Get the registration data
+    my $statement = sprintf("SELECT * from Registration where GUID=%s and PRODUCTID=%s",
+                            $dbh->quote($regGUID),
+                            $dbh->quote($prodID));
```
This could make problems. PRODUCTID is referencing Products.ID which is an autoincrement value.
It is guaranteed that this ID is the same on all SMT server. The PRODUCTDATAID column contains the ID from SCC which is the same on all SMT servers.
But this would make the code a bit more complex of cause.

rjschwei replied:
OK, one of the requirements of the infrastructure is that all SMT servers are setup identically. However, now this explains why smt-list-registrations on one of my test setups did not work as expected, i.e. the product did not show up in the listing, although failover still worked on the client :) .

I will modify the code to take the data from the Products table into account. Such that the PRODUCTID is the appropriate value in the Registration table for that server rather than being the value from the sibling.

If I found the correct connection than the ID column of the Products table matches the PRODUCTID column of the Registration table.

rjschwei committed:
- Product data registration sharing
  + Use the PRODUCTDATAID, which is universally correct as supplied by SCC,
    to properly populate the PRODUCTID of the Registration table.

rjschwei commented:
@mcalmer This does the trick from a functionality point of view, tested on a setup with different PRODUCTID values for the same product and the of smt-list-registrations now looks the same on both systems. Client failover works as expected.

rjschwei committed:
- Product data sharing
  + Consider only 1 value for the foreign table query. Previous implementation
    would have introduced column - value mismatch in the insert statement

rjschwei commented:
@mcalmer also confirmed to work independently by customer

mcalmer commented:
merged and package available at Devel:SMT:SLE-12-SMT. Feel free to start a maintenance update if it is urgent.
